src/main.cpp

```diff
@@ -690,7 +690,22 @@ void GetGlobalStatus()
         GlobalStatusStruct.ERRperday = RoundToString(boincmagnitude * GRCMagnitudeUnit(GetAdjustedTime()),2);
         GlobalStatusStruct.project = msMiningProject;
         GlobalStatusStruct.cpid = GlobalCPUMiningCPID.cpid;
-        GlobalStatusStruct.poll = msPoll;
+
+        std::string sMessageKey = ExtractXML(msPoll, "<MK>", "</MK>");
+        std::string sPollExpiration = ExtractXML(msPoll, "<EXPIRATION>", "</EXPIRATION>");
+        // Alerts are displayed as polls but do not have an expiration
+        if(sPollExpiration.empty())
+        {
+            sPollExpiration = std::to_string(pindexBest->nTime);
+        }
+        if (stoll(sPollExpiration) >= pindexBest->nTime)
+        {
+            GlobalStatusStruct.poll = sMessageKey.substr(0,80);
+        }
+        else
+        {
+            GlobalStatusStruct.poll = "No current polls";
```

Comment:
I do not like this.

Reply:
Could you elaborate? I don't see what's wrong with this.